packages/app-builder/src/services/editor/ast-editor.tsx

```diff
@@ -315,16 +317,56 @@ export function useRootOrAndChildValidation(stringPath: string) {
   ]);
 }
 
-export function useValidationStatus(stringPath: string) {
+export function useValidationStatus(
+  stringPath: string,
+  returnValue?: ReturnValue,
+) {
   const path = usePath(stringPath);
   const evaluationErrors = useEvaluationErrors(stringPath);
   const parentPath = useParentPath(path);
+  const parentNode = useParentAstNode(parentPath);
+  const isDivByZeroField =
+    isDivisionDenominator(parentNode, path) &&
+    !returnValue?.isOmitted &&
+    returnValue?.value === 0;
+
   const parentEvaluationErrors = useParentEvaluationErrors(parentPath);
+  const valueIsNull = !returnValue?.isOmitted && returnValue?.value === null;
   return React.useMemo(() => {
-    return getValidationStatus(
+    return getValidationStatus({
       evaluationErrors,
+      valueIsNull,
+      isDivByZeroField,
       parentEvaluationErrors,
-      parentPath?.childPathSegment,
-    );
-  }, [evaluationErrors, parentEvaluationErrors, parentPath]);
+      pathSegment: parentPath?.childPathSegment,
+    });
```

Comment:
I'm not a super fan of this. If possible, I would prefer to have a logic that create an "artificial" `evaluationError`, so we keep the same logic for `getValidationStatus`.

In the end, a little bit like the `"FUNCTION_ERROR"` implementation

Reply:
Ok, so basically move logic out of getValidationStatus and modify the evaluationErrors instead ?
Get it, that makes sense to me

Reply:
Ok I pushed some changes to the PR - those that are related to this comment are in a separate commit from the other ones (see "PR suggestions 2" below)

Reply:
Should still note, however: currently it's only "parent errors" that lead to a "light-error" coloring, however I'm not really sure where that can actually happen anywhere : I can't see this in any case in the rule builder - perhaps it's a "view only" case ?